libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +82,193 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
+    this.getTriggerButton().nativeElement.focus();
+  }
+
+  resetFocus() {
+    this.focusedIndex = -1;
+  }
+
+  focusItem() {
+    const itemToBeFocused = this.kirbyItems.get(this.focusedIndex);
+    const ionItem = itemToBeFocused.nativeElement.querySelector('ion-item');
+
+    // Look for interactive element within ion-item like toggle or checkbox and set focus if found
+    const firstInteractiveElementWithinItem = this.getFirstInteractiveElement(ionItem);
+    if (firstInteractiveElementWithinItem) {
+      firstInteractiveElementWithinItem['setFocus']();
+    } else {
+      this.focusSelectableItem(ionItem);
+    }
+  }
+
+  private focusSelectableItem(ionItem: HTMLIonItemElement) {
+    const selectableItem: HTMLButtonElement =
+      ionItem.shadowRoot.querySelector('button:not([disabled])');
+    if (selectableItem) {
+      selectableItem.focus();
+    }
+  }
+
+  getTriggerButton(): ElementRef<HTMLButtonElement> {
+    return this.userProvidedButton ?? this.defaultButtonElement;
+  }
 
   public ngAfterViewInit(): void {
-    this.cdf.detectChanges(); // Sets the updated reference for kirby-floating
+    this.cdr.detectChanges(); // Sets the updated reference for kirby-floating
 
     this.zone.runOutsideAngular(() => {
       /*
        * Listen for ionScroll outside of Angular's change detection to
        * avoid a change detection cycle for every scroll-event fired
        */
       this.scrollListenerDisposeFn = this.renderer.listen(document, 'ionScroll', () => {
-        this.floatingDirective.hide();
+        this._floatingMenu.hide();
       });
     });
   }
 
+  ngAfterContentInit(): void {
+    this.setupAccessibilityForItems();
+    this.setupAccesibilityForUserProvidedButton();
+  }
+
+  private setupAccessibilityForItems() {
+    this.kirbyItems.forEach((item) => {
+      item.nativeElement.setAttribute('role', 'menuitem');
+    });
+  }
+
+  menuVisibilityChanged(menuIsShown: boolean) {
+    this.floatingMenuIsShown = menuIsShown;
+    this.getTriggerButton().nativeElement.setAttribute('aria-expanded', menuIsShown + '');
+    if (!menuIsShown) {
+      this.resetFocus();
+      this.focusTriggerButton();
+    }
+  }
+
+  private setupAccesibilityForUserProvidedButton() {
+    if (this.userProvidedButton) {
+      const element = this.userProvidedButton.nativeElement;
+
+      this.setupAriaHasPopup(element);
+      this.setupAriaControls(element);
+      this.setupId(element);
+    }
+  }
+
+  private setupId(button: HTMLButtonElement) {
+    if (!button.id) {
+      button.id = 'userProvidedButton';
+    }
+    this.triggerButtonId = button.id;
+  }
+
+  private setupAriaControls(button: HTMLButtonElement) {
+    if (!button.getAttribute('aria-controls')) {
+      button.setAttribute('aria-controls', this.MENU_ID);
+    }
+  }
+
+  private setupAriaHasPopup(button: HTMLButtonElement) {
+    if (!button.getAttribute('aria-haspopup')) {
+      button.setAttribute('aria-haspopup', 'true');
```

Comment:
Consider using Angular's Renderer2 to avoid the attribute being wiped if the view is re-rendered:
```suggestion
      this.renderer.setAttribute(button, 'aria-haspopup', 'true');
```